src/lib/core/coio.cc

```diff
@@ -159,14 +159,8 @@ int
 coio_connect_timeout(struct ev_io *coio, struct uri *uri, struct sockaddr *addr,
 		     socklen_t *addr_len, ev_tstamp timeout)
 {
-	char host[URI_MAXHOST] = { '\0' };
-	if (uri->host) {
-		snprintf(host, sizeof(host), "%.*s", (int) uri->host_len,
-			 uri->host);
-	}
-	char service[URI_MAXSERVICE];
-	snprintf(service, sizeof(service), "%.*s", (int) uri->service_len,
-		 uri->service);
+	const char *service = (uri->service != NULL ? uri->service : "");
+	const char *host = (uri->host != NULL ? uri->host : "");
```

Comment:
This code was recently moved to applier. Please rebase.

Reply:
fixed